integration/integration_test.go

```diff
@@ -1142,6 +1143,150 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 	}
 }
 
+// testSyncRecordingFailures given a cluster with recording mode set to
+// sync, ensures the session is terminated in case of session recording
+// failures.
+//
+// To simulate failures, we will start an auth server, proxy and ssh server
+// separated. With the session running, we'll stop the auth server, causing any
+// sync recordings to fail. This will works for both, proxy and node recordings.
+func testSyncRecordingFailures(t *testing.T, suite *integrationTestSuite) {
```

Comment:
Can we also get a test (or do we already have one) for when in non-sync mode the session continues?